### DISCUSSION TITLE
2 versions updated but different dependencies

### DISCUSSION THREAD

**Discussion by nntk** [renovatebot/renovate#11525] [Request Help]
I have a gradle.properties file with one gradle plugin and a runtime dependency.
While they are different I get one PR for only updating the spotbugs plugin to the correct version
and another one which updates both version from plugin and runtime dependency like shown below.

```
# renovate: depName=com.github.spotbugs:com.github.spotbugs.gradle.plugin
-spotbugsVersion = 4.7.1
+spotbugsVersion = 4.4.0

# renovate: depName=com.github.spotbugs:spotbugs-annotations
-spotbugsAnnotationsVersion = 4.2.3
+spotbugsAnnotationsVersion = 4.4.0
```

Here is the regexManager that should take care of the comment matching

```
regexManagers: [
    {
      // Gradle - gradle.properties
      // Attach a comment to a dependency version specified in gradle.properties which isn't otherwise picked up by renovate
      fileMatch: ["^gradle\\.properties$"],
      matchStrings: [
        "\\n#\\s+renovate:\\s(depName=(?<depName>.*?))(\\s+datasource=(?<datasource>.*?))?(\\s+versioning=(?<versioning>.*?))?\\s+.*Version\\s+=\\s+(?<currentValue>.*)\\s"
      ],
      datasourceTemplate: "{{#if datasource}}{{datasource}}{{else}}maven{{/if}}",
      versioningTemplate: "{{#if versioning}}{{versioning}}{{else}}gradle{{/if}}",
    }

```

How do I configure this to prevent a wrong version to get updated?

**Comment by viceice** [comment]
your regex is probably too greedy

https://regex101.com/r/xb24uc/1

**Reply by nntk** [reply]
you're right, I had a space at the end that I didn't realize

**Reply by nntk** [reply]
but still updates it, even after the clear cache and deleting of the branch

**Reply by viceice** [reply]
I think `.*Version` is still too greedy, use `[\w]+Version` or something like that

**Reply by viceice** [reply]
https://regex101.com/r/xb24uc/5

**Reply by viceice** [reply]
Please provide the `dependencies with updates` debug message. There you can see the full matching line's
